### ISSUE TITLE
Couldn't connect to Discord via RPC: Error: Could not connect

### ISSUE THREAD

**Issue by power** [iCrawl/discord-vscode#85]
Seem to be getting this error on both clients, canary and stable.

**Comment by iCrawl** [comment]
What OS?
VSCode version?
More than one discord client open?

**Comment by power** [comment]
Windows 10
1.26.1
Only Discord client open

**Comment by power** [comment]
Just after writing this issue seemed to fix, no idea what the issue was. Sorry for wasting your time 👍

**Comment by smt287** [comment]
This problem happened to me. The solution is as follows:
If you are running Discord as an administrator, you must close it in the program settings.
Ex: [https://i.hizliresim.com/NDoXkN.png](url)

**Comment by voloshink** [comment]
@smt287 your link is broken

**Comment by smt287** [comment]
> @smt287 it is broken again 😢

updated again :)

**Comment by Tu4nNguyen** [comment]
Run Vs Code with administrator.

**Comment by Shimeri** [comment]
> 
> 
> Run Vs Code with administrator.

This worked for me

**Comment by asmoaesl** [comment]
I'm getting this error on Linux Manjaro, Code doesn't take running as the super user very well. What is the solution?

**Comment by dutchvanderlinde** [comment]
@asmoaesl Having the same exact issue on Manjaro, I installed it via Snap, AUR, and finally a TAR file and it's still refusing to cooperate. Fun!

**Comment by asmoaesl** [comment]
@dutchvanderlinde I know this is totally off-topic, but I used manjaro for a couple years and tried MX Linux recently, and it is the best distro I've ever used. I say do give it a live try if you're not using Manjaro for any special reason

**Comment by isakal** [comment]
I got this error on elementaryOS version 5.1 Hera.
I installed vscode and discord both through Flatpak and it gave me RPC eror.
So I installed both `.deb` files from their official sites ([vscode](https://code.visualstudio.com/), [discord](https://discordapp.com)). 
After unpacking both files, I started them both and it worked.
 

TL;DR: Install both vscode and discord form their official sites to fix it.

**Comment by DanPHP7** [comment]
> I got this error on elementaryOS version 5.1 Hera.
> I installed vscode and discord both through Flatpak and it gave me RPC eror.
> So I installed both `.deb` files from their official sites ([vscode](https://code.visualstudio.com/), [discord](https://discordapp.com)).
> After unpacking both files, I started them both and it worked.
> 
> TL;DR: Install both vscode and discord form their official sites to fix it.

I got this error on Elementary OS5.1 hera  too, and this worked to me Thaks!!

**Comment by AngelGarcia20** [comment]
I'm using KDE Neon and for some reason I have this error no matter what I do. I installed discord and vscode from the official sites, yet the error persists, anyone else had this issue with KDE too?

**Comment by DiZiNnEs** [comment]
I use Ubuntu 19.10 and I also have this error, downloaded from the off site well it didn’t help

**Comment by fl4m3x** [comment]
I use in Windows 10 (but in my secondary account) and not working... even running as Administrator

**Comment by afcjunior** [comment]
I'm having this issue on macOS catalina. Any idea on how to fix it?

**Comment by RiccardoCafa** [comment]
Same bug here.

Window 10
VSCode v1.43.2

Just installed Discord Presence and didn't worked.
Tried to run VSCode as administrator and still not working. Any solution?

**Comment by isakal** [comment]
@RiccardoCafa After some time i had the same issue.

Solved it by reinstalling Discord Rich Presence in VSCode. From what i understand, Discord is pushing out a lot of internal and UI changes and that usually messes stuff up.

**Comment by zidny123** [comment]
> Run Vs Code with administrator.

thx

**Comment by feelan03** [comment]
Same bug her
Ubuntu 20.04
VsCode 1.44.2

Just installed Discord Presence and didn't worked.

**Comment by Gouthamkc** [comment]
Same i tried to run it as admin

**Comment by danmalloc** [comment]
> Same bug her
> Ubuntu 20.04
> VsCode 1.44.2
> 
> Just installed Discord Presence and didn't worked.

Same for me

**Comment by markkkkas** [comment]
> Same bug her
> Ubuntu 20.04
> VsCode 1.44.2
> 
> Just installed Discord Presence and didn't worked.

install vscode and discord from them original websites

**Comment by sell** [comment]
I am on MAC OS, doesn't work for me. Getting same issue.

**Comment by N04h31** [comment]
> Run Vs Code with administrator.

worked. Thank you.

**Comment by SimonMillerIO** [comment]
1-Windows 10
2- VsCode 1.45
3-Discord , VsCode!

**Comment by AlphaRacer** [comment]
I couldn't get the Rich Presence to show up on my Discord account, even though it said that it had connected. 
I am using MacOS 10.14.
I have the VS Code extension installed and I have the desktop version of discord.
Help would be appreciated!

**Comment by AlbertKang1004** [comment]
I have a same issue with macOS 10.15.4. Visual Studio Code discord rpc is not working.

**Comment by LittleWhole** [comment]
Same issue on pop!_OS 20.04 LTS.

**Comment by Synpeth** [comment]
Same issue on windows 10 and with the latest updates

**Comment by ghaztt** [comment]
i'm running both with administrator and the error continues

**Comment by lukevinicius** [comment]
> Same bug her
> Ubuntu 20.04
> VsCode 1.44.2
> 
> Just installed Discord Presence and didn't worked.

did you manage to solve the problem?

**Comment by Synpeth** [comment]
> Same issue on windows 10 and with the latest updates

I re-installed discord and it worked

**Comment by Arthurdw** [comment]
Waiting a couple of minutes and then reloading discord (`CTRL + R`) should fix this problem.

**Comment by lukegotjellyfish** [comment]
Since I didn't see it mentioned:
- Disabling Discord from being run on start-up can stop a permission error without having to run VSCode with elevated privileges on Win10.

**Comment by CupOfTea134** [comment]
> 
> 
> I couldn't get the Rich Presence to show up on my Discord account, even though it said that it had connected.
> I am using MacOS 10.14.
> I have the VS Code extension installed and I have the desktop version of discord.
> Help would be appreciated!

Have you tried adding VS Code as a game in Settings > Game Activity menu?

**Comment by liondy** [comment]
For those using Ubuntu 20.04 having this issue, try to install discord from .deb and not from snap package. I don't know if something broken if I installed Discord via snap package. Then, you should run your discord (doesn't matter if you start your discord or vscode first), the discord presence will keep reconnecting to discord until it finds discord in your OS. Hope this helps! Feel free to share if any better solution than this.

**Comment by ramishenouda** [comment]
for those using Windows having this issue, try opening discord and vs code as administrator. works with me.

**Comment by kavin25** [comment]
> This problem happened to me. The solution is as follows:
> If you are running Discord as an administrator, you must close it in the program settings.
> Ex: https://erosennin.net/dosyalar/dasd.png

The link is broken😥

**Comment by blackcat-917** [comment]
> This problem happened to me. The solution is as follows:
If you are running Discord as an administrator, you must close it in the program settings.
Ex: https://erosennin.net/dosyalar/dasd.png

This is the image sent by @sametgirginer recovered from archive.org (9 Apr 2019 snapshot)

![image](https://user-images.githubusercontent.com/53786619/96016489-199c3080-0e49-11eb-90f9-69cc59025b1d.png)

**Comment by ErtanOzdemir** [comment]
Here is the solution for Ubuntu Users;

If you installed VSCode and Discord via Ubuntu Software Store you need to uninstall them.
``` 
snap list
sudo snap remove code
sudo snap remove discord
```
After that go to official web pages of Discord and VsCode. Download .deb files to your computer. Install these .deb packages.
``` 
sudo dpkg -i <package name>
```
After installed VsCode and Discord, open VsCode and install Discord Presence plugin. 
Then Open Discord and go to <strong> User Settings > Game Activity </strong> page.
Turn On "<strong>Display currently running game as a status message.</strong>" setting.
It must be work.

If you are still encountering the error, press <strong>Add it!</strong>  in "<strong>Not seeing your game? </strong>" section then try to find your VS Code.

**Comment by tanjilk** [comment]
> Here is the solution for Ubuntu Users;
> 
> If you installed VSCode and Discord via Ubuntu Software Store you need to uninstall them.
> 
> ```
> snap list
> sudo snap remove code
> sudo snap remove discord
> ```
> 
> After that go to official web pages of Discord and VsCode. Download .deb files to your computer. Install these .deb packages.
> 
> ```
> sudo dpkg -i <package name>
> ```
> 
> After installed VsCode and Discord, open VsCode and install Discord Presence plugin.
> Then Open Discord and go to ** User Settings > Game Activity ** page.
> Turn On "**Display currently running game as a status message.**" setting.
> It must be work.
> 
> If you are still encountering the error, press **Add it!** in "**Not seeing your game? **" section then try to find your VS Code.

Thanks a lot, it worrks on my Ubuntu 20.04
I installed from the discord official server and vs code too.

**Comment by Artivain** [comment]
Happened to me today, I closed Discord and VS Code from task manager and restarted them normally. Worked perfectly fine.

**Comment by Dito2007baik** [comment]
So i was writing some code and when i tried this extension, the first time it worked but tomorrow for some reason it didn't work. For the details are
Discord : Stable 73785 (af82041)
VS Code : 1.52
Windows : Version 20H2

So the problem was i refreshed discord after vscode opened and already connected to discord but for some reason it kept showing the ? symbol instead . Yes i already connected to discord here is some proof. 
![Proof](https://user-images.githubusercontent.com/45822792/102680986-0f533580-41f0-11eb-8244-6cdbc63c2d0a.png)
So i have tried all of these methods from top to bottom including the one highlighted in the vs marketplace and @Artivain 's method and i didn't ran any of those apps (both discord and vscode) with administrative privileges. And it didn't work. Am I doing something wrong or is there any other fixes that i can try?

**Comment by lxwulf** [comment]
Same Problem here except that VS Code even doesn't show up in discord status. VS Code says it is connected to discord via RPC...

Maybe something we missed in the updates of VS Code or discord? 🤔

VS Code Insiders: 1.53.0
Discord: Stable 73806 (0c3eadc)
Windows: Windows 10 64-Bit (10.0.19042)

**Comment by FrostyyJeps** [comment]
> This problem happened to me. The solution is as follows:
> If you are running Discord as an administrator, you must close it in the program settings.
> Ex: https://erosennin.net/dosyalar/dasd.png

This fixed it for me. Thanks alot man 🤠 

(link is still broke 🚮)

**Comment by lxwulf** [comment]
What do you mean with program settings? I also don't run discord as administrator....

**Comment by bentoco** [comment]
> Here is the solution for Ubuntu Users;
> 
> If you installed VSCode and Discord via Ubuntu Software Store you need to uninstall them.
> 
> ```
> snap list
> sudo snap remove code
> sudo snap remove discord
> ```
> 
> After that go to official web pages of Discord and VsCode. Download .deb files to your computer. Install these .deb packages.
> 
> ```
> sudo dpkg -i <package name>
> ```
> 
> After installed VsCode and Discord, open VsCode and install Discord Presence plugin.
> Then Open Discord and go to ** User Settings > Game Activity ** page.
> Turn On "**Display currently running game as a status message.**" setting.
> It must be work.
> 
> If you are still encountering the error, press **Add it!** in "**Not seeing your game? **" section then try to find your VS Code.

This works very well. Thanks a lot!

**Comment by lxwulf** [comment]
I hope it wasn't this failure, but I think I forgot to set up this toggle here...._at least on my fedora system_

[Screenshot-20210127-165443.png](https://postimg.cc/D8ggskXY)

I would look again to windows maybe the same thing....

**Comment by yezz123** [comment]
The same problem for me he want work on Ubuntu 20.04.2LTS.

![Screenshot from 2021-02-06 10-20-04](https://user-images.githubusercontent.com/52716203/107114275-e866c680-6864-11eb-8687-bb7d008d7a31.png)

**Comment by tanjilk** [comment]
Try to do what @bentoco did. I installed from the official web pages of
discord and visual studio code too it worked for me.

Yasser Tahiri <notifications@github.com> escreveu no dia sábado, 6/02/2021
à(s) 09:21:

**Comment by Ultimus2935** [comment]
for me the fix was easy, i was on discord canary and it gave me the "coudnt connect" error. then i uninstalled that and reinstalled the normal stable version and it fixed it. it is working normally since then

**Comment by EugeneGohh** [comment]
By the way, I fixed it by turning on **Game Activity** on Discord.

![Untitled](https://user-images.githubusercontent.com/64187129/111256992-443b2100-8655-11eb-8201-e0c766aa81de.png)

**Comment by Maritsu** [comment]
@EugeneGohh Bruh, was it rly that simple? That should be pinned or mentioned in README

**Comment by icarusgk** [comment]
I solved it by opening up Discord again

**Comment by lxwulf** [comment]
@Maritsu yeah it could be I had the same "error". I agree to give the users a tip in the README. 🤔

**Comment by EugeneGohh** [comment]
> @EugeneGohh Bruh, was it rly that simple? That should be pinned or mentioned in README

Are you having the same issue when you downloaded Discord Presence?

**Comment by Maritsu** [comment]
> > @EugeneGohh Bruh, was it rly that simple? That should be pinned or mentioned in README
> 
> Are you having the same issue when you downloaded Discord Presence?

@EugeneGohh I just didn't notice that Game Activity was turned off

**Comment by Rhydium** [comment]
I solved this problem on Windows 10 Pro 20H2 by closing my Discord and running the DiscordSetup.exe again.
(Fixed this specific error for me and being stuck on "Connecting to Discord")

**Comment by thanhgaming5550** [comment]
> Run Vs Code with administrator.

worked for me
VSCode : Admin
Discord : Without Admin

**Comment by Bravehard700** [comment]
Thank you very much it worked 😁😁

**Comment by LogicApples** [comment]
> > Run Vs Code with administrator.
> 
> worked for me VSCode : Admin Discord : Without Admin

question. What if you can't run VSCode as an Admin?

**Comment by ksteflovic** [comment]
> Run Vs Code with administrator.

This helped me. Thank you

**Comment by bruhman-del** [comment]
> What OS? VSCode version? More than one discord client open?

what to do if i am on linux :cry:

**Comment by lxwulf** [comment]
Yes, still need to know which version or OS you have installed and also from which source -> packet manager? flatpak? snap?

Greets LxWulf

**Comment by lxwulf** [comment]
> > What OS? VSCode version? More than one discord client open?
> 
> what to do if i am on linux :cry: 
> 

Yes, still need to know which version or OS you have installed and also from which source -> packet manager? flatpak? snap?

Greets LxWulf

**Comment by bschaatsbergen** [comment]
Still running in to this..

![image](https://user-images.githubusercontent.com/58337159/165259076-46518acc-8dbf-4796-b3da-801fd4d5012e.png)

**Comment by PurplProto** [comment]
I get this issue on Windows sometimes. What works for me (no, not run as admin, you don't and shouldn't need that), was:
1. Open Discord user settings
2. Click Activity Status
3. Ensure "Display current activity as a status message" is enabled
4. Right-click Discord in the system tray
5. Click quit 
6. Open task manager
7. Select the "Details" tab
8. Search for any remaining Discord processes
9. Right-click each and click "End process tree"
10. Start Discord
11. Disable the Discord presence extension in VSCode
12. Reload VSCode when it requests it
13. Re-enable the Discord presence extension 
14. Observe no error appearing anymore 🙌

Hope that helps 😊

**Comment by debanath** [comment]
> I get this issue on Windows sometimes. What works for me (no, not run as admin, you don't and shouldn't need that), was:
> 
> 1. Open Discord user settings
> 2. Click Activity Status
> 3. Ensure "Display current activity as a status message" is enabled
> 4. Right-click Discord in the system tray
> 5. Click quit
> 6. Open task manager
> 7. Select the "Details" tab
> 8. Search for any remaining Discord processes
> 9. Right-click each and click "End process tree"
> 10. Start Discord
> 11. Disable the Discord presence extension in VSCode
> 12. Reload VSCode when it requests it
> 13. Re-enable the Discord presence extension
> 14. Observe no error appearing anymore 🙌
> 
> Hope that helps 😊

It did help. "The Ultimate guide to fix the issue"

**Comment by kiosion** [comment]
> I get this issue on Windows sometimes. What works for me (no, not run as admin, you don't and shouldn't need that), was:
> 
> 1. Open Discord user settings
> 2. Click Activity Status
> 3. Ensure "Display current activity as a status message" is enabled
> 4. Right-click Discord in the system tray
> 5. Click quit
> 6. Open task manager
> 7. Select the "Details" tab
> 8. Search for any remaining Discord processes
> 9. Right-click each and click "End process tree"
> 10. Start Discord
> 11. Disable the Discord presence extension in VSCode
> 12. Reload VSCode when it requests it
> 13. Re-enable the Discord presence extension
> 14. Observe no error appearing anymore raised_hands
> 
> Hope that helps blush

Didn't think to check for Discord processes after closing it, but can confirm this worked, there were about 10 dead Discord-related processes still open! And just to chime in as well, you absolutely shouldn't run Discord or VSCode as admin/root as a troubleshooting step

**Comment by sharpsailor** [comment]
> > I get this issue on Windows sometimes. What works for me (no, not run as admin, you don't and shouldn't need that), was:
> > 
> > 1. Open Discord user settings
> > 2. Click Activity Status
> > 3. Ensure "Display current activity as a status message" is enabled
> > 4. Right-click Discord in the system tray
> > 5. Click quit
> > 6. Open task manager
> > 7. Select the "Details" tab
> > 8. Search for any remaining Discord processes
> > 9. Right-click each and click "End process tree"
> > 10. Start Discord
> > 11. Disable the Discord presence extension in VSCode
> > 12. Reload VSCode when it requests it
> > 13. Re-enable the Discord presence extension
> > 14. Observe no error appearing anymore raised_hands
> > 
> > Hope that helps blush
> 
> Didn't think to check for Discord processes after closing it, but can confirm this worked, there were about 10 dead Discord-related processes still open! And just to chime in as well, you absolutely shouldn't run Discord or VSCode as admin/root as a troubleshooting step

I did try using above steps but no changes ,it still shows the error
currently I am on Ubuntu 22.04.1 LTS x86_64.

**Comment by cjrobbertse** [comment]
It would also seem that you need to have the discord app installed locally, but no one mentioned that... 😅

**Comment by devluisg** [comment]
> This problem happened to me. The solution is as follows: If you are running Discord as an administrator, you must close it in the program settings. Ex: https://imgur.com/a/vlNITQ2 

Okay, I'm just dropping by to give my report. It may help someone.

The same thing was happening to me - the extension only worked when I installed it, but the next day it would stop working even though nothing had changed in the system. However, when I came to look for a solution, I saw the comment from @sametgirginer and I tested it and it worked. However, I was intrigued that both the extension description and other comments were saying that this step wasn't necessary. Seeing this, I was sure that the problem was with VSCode, so I decided to do some research and found out that VSCode has two installers: the User Installer and the System Installer. It turns out that when you use the main button to install, it ends up downloading the User version. I uninstalled my VSCode and decided to download and install the System version instead, and I haven't had the problem for three days now. I hope this report can help more people.

In short, try installing the System Version instead of the User Version of VScode.

**Comment by MrS4w** [comment]
I just opened VS Code as administrator and it fixed to me

**Comment by paoloose** [comment]
It might be obvious, but make sure your status is not "Invisible". Mine was.

**Comment by RH203** [comment]
> Windows

It did help. I use windows OS ✌️

**Comment by Akronae** [comment]
For my case I had several instances of Discord running without knowing it (nothing showing in the icon tray when quitting).
I had to end all Discord processes in Task Manager, open discord again, and reconnect from VSCode.

**Comment by blackdragoon26** [comment]
> I get this issue on Windows sometimes. What works for me (no, not run as admin, you don't and shouldn't need that), was:
> 
> 1. Open Discord user settings
> 2. Click Activity Status
> 3. Ensure "Display current activity as a status message" is enabled
> 4. Right-click Discord in the system tray
> 5. Click quit
> 6. Open task manager
> 7. Select the "Details" tab
> 8. Search for any remaining Discord processes
> 9. Right-click each and click "End process tree"
> 10. Start Discord
> 11. Disable the Discord presence extension in VSCode
> 12. Reload VSCode when it requests it
> 13. Re-enable the Discord presence extension
> 14. Observe no error appearing anymore 🙌
> 
> Hope that helps 😊

 Thanx alot, so main problem was that unnecessary service of dc running in bg.....

**Comment by timpo14** [comment]
> Also worth checking that you dont have rich presence disabled in vscode: ![image](https://private-user-images.githubusercontent.com/6438565/282854034-426ad250-74d2-49ab-b6bc-30fb2eefa65a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMxNDA3NTQsIm5iZiI6MTcyMzE0MDQ1NCwicGF0aCI6Ii82NDM4NTY1LzI4Mjg1NDAzNC00MjZhZDI1MC03NGQyLTQ5YWItYjZiYy0zMGZiMmVlZmE2NWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDhUMTgwNzM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODMyMjRhMjgzZTI3NmE3YzRhOTg0MTk4MzU4ZTMwMzU1YjcwZWNkZTBkNWJmOGZhMTcwZWYxMDg1NjVkYjM0NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wA_HbJrPcMbf9NHzE15WybpQqm-v4GnLDokgOnM3V-s)

Why my Discord missing this? I dont know, maybe i'm blind, but i'm don't see it

**Comment by dreamy32** [comment]
My straightforward way to "fix", which is way faster than some of the solutions proposed above:

Steps:
1. **CTRL+R** on Discord to refresh
2.  Click on **Connect to Discord**

**Comment by Lelin07** [comment]
**Check whether your status is invisible on discord**

It might sound dumb but turning the discord status from invisible to ideal solved my problem.

**Comment by Jervi-sir** [comment]
I found this in their extensions vscode details, 
issue got fixed, after I installed Discord from their site not from snap
![Image](https://github.com/user-attachments/assets/196148a3-9756-4064-958a-72b139f82759)

**Comment by Jonhvmp** [comment]
The solution for me was to delete the extension as it is useless, it only serves to take up space.

**Comment by eldessaDev** [comment]
> This problem happened to me. The solution is as follows: If you are running Discord as an administrator, you must close it in the program settings. Ex: https://imgur.com/a/vlNITQ2

WORK!

**Comment by Malia-lai** [comment]
Need discord downloaded locally and openned at least in the background
